src/exchange_adapters/bitstamp.ts

```diff
@@ -0,0 +1,82 @@
+import { Exchange } from '../utils'
+import { BaseExchangeAdapter, ExchangeAdapter, ExchangeDataType, Ticker, Trade } from './base'
+
+export class BitstampAdapter extends BaseExchangeAdapter implements ExchangeAdapter {
+  baseApiUrl = 'https://www.bitstamp.net/api/v2/'
+  readonly _exchangeName = Exchange.BITSTAMP
+  readonly _certFingerprint256 =
+    '40:3E:06:2A:26:53:05:91:13:28:5B:AF:80:A0:D4:AE:42:2C:84:8C:9F:78:FA:D0:1F:C9:4B:C5:B8:7F:EF:1A'
+
+  private static readonly tokenSymbolMap = BitstampAdapter.standardTokenSymbolMap
+
+  protected generatePairSymbol(): string {
+    const base = BitstampAdapter.tokenSymbolMap.get(this.config.baseCurrency)
+    const quote = BitstampAdapter.tokenSymbolMap.get(this.config.quoteCurrency)
+    return `${base}${quote}`
+  }
+
+  async fetchTicker(): Promise<Ticker> {
+    const tickerJson = await this.fetchFromApi(
+      ExchangeDataType.TICKER,
+      `ticker/${this.pairSymbol.toLowerCase()}`
```

Comment:
nit: what do you think of doing the .toLowerCase() in `generatePairSymbol` instead? sounds like it'd make more sense there

Reply:
agree!!